### PR TITLE
style: update linting to support paver sass compile

### PR DESCRIPTION
**Pull request by leangseu-edx**
TLDR; Make distinction between rgb and rgba because `edx-platform` doesn't support it. Update `stylelint` to maintain it.

## Description

When use sass compile on `edx-platform`, some of the modern syntax isn't supported. For example, `rgb(0 0 0 / 50%)` need to be convert to `rgba(0, 0, 0, 0.5)`.   See [color-function-notation](https://stylelint.io/user-guide/rules/list/color-function-notation/]

### Deploy Preview

Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Review by adamstankiewicz — APPROVED** [review]
LGTM! :shipit:
